internal/cmd/launcher.go

```diff
@@ -216,8 +217,23 @@ func newK6BuildProvisioner(gs *state.GlobalState) provisioner {
 }
 
 func (p *k6buildProvisioner) provision(deps k6deps.Dependencies) (commandExecutor, error) {
+	buildSrv := p.gs.Flags.BuildServiceURL
+	buildSrvURL, err := url.Parse(buildSrv)
+	if err != nil {
+		return nil, fmt.Errorf("invalid URL to binary provisioning build service: %w", err)
+	}
+
+	catalog := state.CloudExtensionsCatalog
+	if p.gs.Flags.EnableCommunityExtensions {
+		catalog = state.CommunityExtensionsCatalog
+	}
+	buildSrv = buildSrvURL.JoinPath(catalog).String()
```

Comment:
We might consider moving this logic into a dedicated function that accepts as input a flagset and it returns the URL or an error.

In this way, it's easier to test.